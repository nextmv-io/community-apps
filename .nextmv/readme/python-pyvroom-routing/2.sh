cat input.json | docker run -i --rm \
-v $(pwd):/app ghcr.io/nextmv-io/runtime/pyomo:latest \
sh -c 'python3 /app/main.py'
