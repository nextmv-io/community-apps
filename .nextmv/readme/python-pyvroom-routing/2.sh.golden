{
  "solutions": [
    {
      "vehicles": [
        {
          "id": "vehicle-1",
          "route_travel_duration": 1126,
          "route": [
            {
              "stop": {
                "id": "vehicle-1_start",
                "location": {
                  "lon": 7.625,
                  "lat": 51.9622
                }
              },
              "type": "start",
              "arrival": 0,
              "duration": 0,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "location-6",
                "location": {
                  "lon": 7.6292,
                  "lat": 51.9577
                },
                "quantity": 1
              },
              "type": "stop",
              "arrival": 129,
              "duration": 129,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "location-10",
                "location": {
                  "lon": 7.6299,
                  "lat": 51.9582
                },
                "quantity": 1
              },
              "type": "stop",
              "arrival": 138,
              "duration": 138,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "location-7",
                "location": {
                  "lon": 7.6385,
                  "lat": 51.9694
                },
                "quantity": 1
              },
              "type": "stop",
              "arrival": 320,
              "duration": 320,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "location-8",
                "location": {
                  "lon": 7.6628,
                  "lat": 51.9616
                },
                "quantity": 1
              },
              "type": "stop",
              "arrival": 594,
              "duration": 594,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "location-2",
                "location": {
                  "lon": 7.6353,
                  "lat": 51.9584
                },
                "quantity": 1
              },
              "type": "stop",
              "arrival": 864,
              "duration": 864,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "vehicle-1_end",
                "location": {
                  "lon": 7.625,
                  "lat": 51.9622
                }
              },
              "type": "end",
              "arrival": 1126,
              "duration": 1126,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            }
          ]
        },
        {
          "id": "vehicle-2",
          "route_travel_duration": 2350,
          "route": [
            {
              "stop": {
                "id": "vehicle-2_start",
                "location": {
                  "lon": 7.625,
                  "lat": 51.9622
                }
              },
              "type": "start",
              "arrival": 0,
              "duration": 0,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "location-3",
                "location": {
                  "lon": 7.6233,
                  "lat": 51.9638
                },
                "quantity": 1
              },
              "type": "stop",
              "arrival": 372,
              "duration": 372,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "location-9",
                "location": {
                  "lon": 7.6217,
                  "lat": 51.9667
                },
                "quantity": 1
              },
              "type": "stop",
              "arrival": 666,
              "duration": 666,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "location-4",
                "location": {
                  "lon": 7.6163,
                  "lat": 51.9635
                },
                "quantity": 1
              },
              "type": "stop",
              "arrival": 755,
              "duration": 755,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "location-5",
                "location": {
                  "lon": 7.6045,
                  "lat": 51.9509
                },
                "quantity": 1
              },
              "type": "stop",
              "arrival": 962,
              "duration": 962,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "location-1",
                "location": {
                  "lon": 7.6228,
                  "lat": 51.9624
                },
                "quantity": 1
              },
              "type": "stop",
              "arrival": 1737,
              "duration": 1737,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            },
            {
              "stop": {
                "id": "vehicle-2_end",
                "location": {
                  "lon": 7.625,
                  "lat": 51.9622
                }
              },
              "type": "end",
              "arrival": 2350,
              "duration": 2350,
              "setup": 0,
              "service": 0,
              "waiting_time": 0
            }
          ]
        }
      ],
      "unplanned": []
    }
  ],
  "statistics": {
    "result": {
      "custom": {
        "solution_found": true,
        "activated_vehicles": 2,
        "max_route_duration": 2350,
        "max_stops_in_vehicle": 5,
        "min_stops_in_vehicle": 5
      },
      "duration": 0.123,
      "value": 3476
    },
    "run": {
      "duration": 0.123
    },
    "schema": "v1"
  }
}
